src/analysis/complex/conformal.lean

```diff
@@ -0,0 +1,136 @@
+/-
+Copyright (c) 2021 Yourong Zang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yourong Zang
+-/
+import analysis.complex.isometry
+import analysis.calculus.conformal
+
+/-!
+# Complex Conformal Maps
+
+We prove the sufficient and necessary conditions for a complex function to be conformal.
+
+## Main results
+
+* `is_conformal_map_complex_linear`: a nonzero complex linear map into an arbitrary complex
+                                     normed space is conformal.
+* `is_conformal_map_complex_linear_conj`: the composition of a nonzero complex linear map with
+                                          `conj` is complex linear.
+* `is_complex_or_conj_complex_linear_iff_is_conformal_map`: a real linear map between the complex
+                                                            plane is conformal iff it's complex
+                                                            linear or the composition of
+                                                            some complex linear map and `conj`.
+
+## Warning
+
+Antiholomorphic functions such as the complex conjugate are considered as conformal functions in
+this file.
+-/
+
+noncomputable theory
+
+open complex continuous_linear_map
+
+section conformal_into_complex_normed
+
+variables {E : Type*} [normed_group E] [normed_space ℝ E] [normed_space ℂ E]
+  [is_scalar_tower ℝ ℂ E] {z : ℂ} {g : ℂ →L[ℝ] E} {f : ℂ → E}
+
+lemma is_conformal_map_complex_linear
+  {map : ℂ →L[ℂ] E} (nonzero : map ≠ 0) : is_conformal_map (map.restrict_scalars ℝ) :=
+begin
+  have minor₀ : ∀ (x : ℂ), x = x • 1 := λ x, by rw [smul_eq_mul, mul_one],
+  have minor₁ : ∥map 1∥ ≠ 0,
+  { contrapose! nonzero with w,
+    ext1,
+    rw [continuous_linear_map.zero_apply],
+    exact norm_eq_zero.mp w, },
+  refine ⟨∥map 1∥, minor₁, _⟩,
+  let li' := ∥map 1∥⁻¹ • map,
+  have key : ∀ (x : ℂ), ∥li' x∥ = ∥x∥,
+  { intros x,
+    simp only [li', continuous_linear_map.smul_apply],
+    nth_rewrite 0 [minor₀ x],
+    simp only [map.map_smul, norm_smul, normed_field.norm_inv, norm_norm],
+    field_simp [minor₁], },
+  let li : ℂ →ₗᵢ[ℝ] E := ⟨li', key⟩,
+  have minor₂ : (li : ℂ → E) = li' := rfl,
+  refine ⟨li, _⟩,
+  ext1,
+  simp only [minor₂, li', pi.smul_apply, continuous_linear_map.smul_apply, smul_smul],
+  field_simp [minor₁],
+end
+
+lemma is_conformal_map_conj : is_conformal_map conj_cle.to_continuous_linear_map :=
+⟨1, one_ne_zero, conj_lie.to_linear_isometry, by rw one_smul; refl⟩
```

Comment:
Can you please add a lemma (back in `analysis/normed_space/conformal_linear_map` saying that an isometry is a conformal map, then deduce this from that?

Reply:
Done. But I changed the lemma into `lemma is_conformal_map_conj : is_conformal_map (conj_lie : C ->L[R] C)` so that the proof would be super short. And all other lemmas work without any modification, probably because `(conj_lie : C ->L[R] C)` and `(conj_cle : C ->L[R] C)` are the same thing in Lean.

Reply:
Great!  I agree that the `conj_lie` version is more elegant.  Could you please also move this lemma to the start of the file, since it doesn't really belong to the `conformal_into_complex_normed` section.

Reply:
Done